src/utils/PrivateRoute.js

```diff
@@ -1,15 +1,30 @@
 import React, { useContext } from "react";
-import { Redirect, Route } from "react-router-dom";
+import { Redirect, Route, Link } from "react-router-dom";
 
 import { IsAuthContext } from "./AuthContext";
 
 function PrivateRoute({ children, ...restOfProps }) {
-  const isAuth = true
-  
-  return isAuth ? 
-    (<Route {...restOfProps}>{children}</Route>)
-       :
-    (<Redirect exact to="/" /> )
+  const { isLoggedIn } = useContext(IsAuthContext);
+  const { isAuth } = useContext(IsAuthContext);
+
+  isLoggedIn();
+
+  if (!isAuth) {
+    return (
+      <p>
+        You need to be logged in. Please login{" "}
+        <Link exact to="/">
+          here
+        </Link>
+      </p>
+    );
+  }
+
+  return isAuth ? (
+    <Route {...restOfProps}>{children}</Route>
+  ) : (
+    <Redirect exact to="/" />
```

Comment:
This `redirect` will never hit.

You already have this condition `if (!isAuth) {`, which means "if the user is not authenticated than go inside of this `if` block an `return ` the code.

And if the user IS authenticated, that means they can `route` so this code will be executed   `<Route {...restOfProps}>{children}</Route>`


Either have:
```javascript

return  isAuth ?  <Route {...restOfProps}>{children}</Route> : <Redirect exact to="/" />
```
or like this:
```javascript

return isAuth ? (
    <Route {...restOfProps}>{children}</Route>
  ) : (
    <p>
        You need to be logged in. Please login{" "}
        <Link exact to="/">
          here
        </Link>
      </p>
)
```